### DISCUSSION TITLE
SpiceDB for generic graph queries

### DISCUSSION THREAD

**Discussion by dberardo-com** [authzed/spicedb#2039] [Q&A]
given that SpiceDB is backed by a graph model, would it be possible to run graph-like queries ,e.g. using Cypher language ?

i mean, one could surely use it as an authorization framework, but why limiting it to it ?

my use case: i am evaluating using Apache AGE on my postgresql DB to run graph based queries on my user relationships. naturally i could use spiceDB for authorization as well, but that would mean having to duplicate data between the 2 engines, since Apache AGE also uses its custom graph model.

**Comment by vroldanbet** [comment]
@dberardo-com the short answer is no in its current form. Even though SpiceDB's relationship model represents a graph, the data isn't queried using the access patterns of a graph database, so it has very little to gain from using a graph database.

**Comment by tstirrat15** [comment]
@dberardo-com to add a little to this - SpiceDB is highly-optimized around three queries (CheckPermission, LookupResources, LookupSubjects) and its behavior and data model reflect that. It isn't particularly good at generalized graph walks. It also isn't suited for "normal" database behaviors like sorting, filtering, and pagination, and you can only store edges, and those edges can't carry metadata in their current form. I don't think that it would work particularly well as a general-purpose graph database and we don't currently have plans to move it in that direction - our focus is on authorization.
